interface-definitions/service-zabbix-agent.xml.in

```diff
@@ -0,0 +1,181 @@
+<?xml version="1.0"?>
+<interfaceDefinition>
+  <node name="service">
+    <children>
+      <node name="zabbix-agent" owner="${vyos_conf_scripts_dir}/service_zabbix_agent.py">
+        <properties>
+          <help>Zabbix-agent settings</help>
+          <priority>380</priority>
+        </properties>
+        <children>
+          <leafNode name="directory">
+            <properties>
+              <help>Folder containing individual Zabbix-agent configuration files</help>
+              <constraint>
+                <validator name="file-path" argument="--directory"/>
+              </constraint>
+            </properties>
+          </leafNode>
+          <node name="limits">
+            <properties>
+              <help>Limit settings</help>
+            </properties>
+            <children>
+              <leafNode name="buffer-send">
+                <properties>
+                  <help>Do not keep data longer than N seconds in buffer</help>
```

Comment:
What exactly does this option do? If it's set to 5 seconds, does the agent send the buffered data every 5 seconds, or does it discard data that survives in a buffer for longer than 5 seconds?

If the former, a more fitting name would be `buffer-flush-interval` or similar. If the latter, something like `buffer-lifetime`.

Also, since there are at least two options related to buffers, maybe make `buffer` a parent node and put this and buffer size under it?

Reply:
> What exactly does this option do? If it's set to 5 seconds, does the agent send the buffered data every 5 seconds, or does it discard data that survives in a buffer for longer than 5 seconds?

The time interval in seconds which determines how often values are sent from the buffer to Zabbix server. Note, that if the buffer is full, the data will be sent sooner.

Reply:
> Also, since there are at least two options related to buffers, maybe make `buffer` a parent node and put this and buffer size under it?

I use common node `limits`
```
set service zabbix-agent limits buffer-send '8'
set service zabbix-agent limits buffer-size '120'
```
Most likely will be extended with other values in the future.

Reply:
`buffer-send` replaced to `set service zabbix-agent limits buffer-flush-interval x`